### ISSUE TITLE
[Addition] [1.12.2] What about CleanRoomMC 

### ISSUE THREAD

**Issue by G1ll0u** [TheUsefulLists/UsefulMods#320]
### Checklist

- [X] I've checked that my mod suggestion hasn't already been added to one of the lists.
- [X] I've checked the latest issues to make sure that this mod hasn't already been suggested.

### What is your mod's name?

CleanRoomMC

### What is your mod's main version?

1.12.X

### What is your mod's type?

Preformance

### What side dose your mod need to run on?

Both

### Input a link to your suggested mod.

https://github.com/CleanroomMC

### What is your reasoning for including this mod?

It's not a mod, It's a completely new, java 21, JWJGL3 forge mod loader for 1.12.2.
I migrated my whole personal dirty modpack to CleanRoomMC very easily (just copying my mods and configs). Loading times are divided by 2 and I see some FPS gain.
Even Optifine works ! Even joining to my forge 2860 works !

Read carefully how to install it [here](https://github.com/CleanroomMC/)!

And note that this software requires two mods to work properly with most existing mods :
https://github.com/CleanroomMC/Fugue
https://github.com/CleanroomMC/Scalar/
Remove !configAnytime and !MixinBooter after installing it (not game-breaking if you forget)

I think there should be a new section under performance named "modloader optimizations" or something like that.

Have a great day !

**Comment by Nekonaro** [comment]
In my case, it gave a small performance boost due to the newer version of java. The main point of this mod is that it allows you to use a new garbage collector, such as The Z Garbage Collector

**Comment by G1ll0u** [comment]
Exactly, but it's not the main point. Note that this is the continuation of Forge 1.12.2, I tried ZGC but java 21 G1GC is sufficient with my ryzen 9 3900x (12cores). (personnaly using xms6G and xmx 12G with my 135 active mods)

<details><summary>flags</summary>-XX:AllocatePrefetchStyle=1 -XX:-ZProactive -XX:+UnlockExperimentalVMOptions -XX:+UnlockDiagnosticVMOptions -XX:+AlwaysActAsServerClassMachine -XX:+AlwaysPreTouch -XX:+DisableExplicitGC -XX:+UseNUMA -XX:NmethodSweepActivity=1 -XX:ReservedCodeCacheSize=400M -XX:NonNMethodCodeHeapSize=12M -XX:ProfiledCodeHeapSize=194M -XX:NonProfiledCodeHeapSize=194M -XX:-DontCompileHugeMethods -XX:MaxNodeLimit=240000 -XX:NodeLimitFudgeFactor=8000 -XX:+UseVectorCmov -XX:+PerfDisableSharedMem -XX:+UseFastUnorderedTimeStamps -XX:+UseCriticalJavaThreadPriority -XX:ThreadPriorityPolicy=1 -XX:+UseLargePages
</details>

I stole these flags here I stole those flags [here](https://github.com/brucethemoose/Minecraft-Performance-Flags-Benchmarks)
90-120 FPS in complex scene without shaders and **without VintageFix or any performance mod incompatible with Optifine**
30-60 FPS depending on scene complexity with Optifine + shaders

Not feeling any GC frame lost.

Before, I was at 20-40 FPS in complex scene with big spikes and now it seems really smoother. I think it is caused by LWJGL3 but I'm absolutely not an expert.

It decreased 1/3 of my loading time.
99% of my old forge mods work.

So I think it would be really cool, before everything about 1.12.2, to talk about this new mod loader. Because more people use it, more it'll be popular and we may, one day, have an Iris 1.12.2... Imagine...

Have a lovely day.
